hphp/runtime/vm/jit/vasm-util.cpp

```diff
@@ -29,7 +29,12 @@ bool is_nop(const copy& i) { return i.s == i.d; }
 bool is_nop(const copy2& i) { return i.s0 == i.d0 && i.s1 == i.d1; }
 
 // movb r,r is a nop, however movl is not since it zeros upper bits.
-bool is_nop(const movb& i) { return i.s == i.d; }
+bool is_nop(const movb& i) {
+  // On PPC64 movb r,r cannot be considered a trivial nop since it zeros
```

Comment:
> If you need to copy 8 bits and zero all the upper bits, you need to lower to movzbq{s, tmp}; movtqb{tmp, d} instead of to movb{s, d}. Is that what you want?

We are lowering those instructions to `movb` because on our implementation we zero's all upper bits on mov. I guess I can lower `movb` to `movzbq` instead.

The real problem is in some lowers we do. Let me show a example with our `xorb`:

``` 
void lowerForPPC64(Vout& v, xorb& inst) { 
  Vreg tmp1 = v.makeReg(), tmp2 = v.makeReg();
  v << movb{inst.s0, tmp1};  
  v << movb{inst.s1, tmp2}; 
  v << xorq{tmp1, tmp2, Reg64(inst.sf)}; 
}
```
The problem is if we have `r(dst) = r(src)` those `movb` will be replaced by a `nop` ending up with wrong values on registers when `xorq` is executed. I am thinking here why we really need those moves.